static/js/spamReferralsBlocker.js

```diff
@@ -0,0 +1,16 @@
+import spammers from 'raw-loader!./spammers.txt';
+
+window.isSpamReferral = function () {
+  const list = spammers.split('\n');
+  const currentReferral = document.referrer;
+  if (currentReferral) {
+    list.unshift('oddsite.hexxie.com', 'localhost');
+    for (const spammer of list) {
+      if (spammer &&
+          currentReferral.toLowerCase().indexOf(spammer.toLowerCase()) !== -1) {
```

Comment:
Yep, I was using that originally, but because some browsers don't support it (e.g. older Android) it requires a polyfill (in `init.js`, we're using `core-js/shim`). Since this script is blocking and executed in the `head`, it seemed silly to pull in more JS than necessary, so I just used the older syntax.